ethcore/evm/src/interpreter/shared_cache.rs

```diff
@@ -36,9 +37,14 @@ impl MallocSizeOf for Bits {
 	}
 }
 
+#[derive(MallocSizeOf, Clone)]
+struct CacheItem {
+	jump_destination: Bits,
+	sub_entrypoint: Bits,
+}
 /// Global cache for EVM interpreter
 pub struct SharedCache {
-	jump_destinations: Mutex<MemoryLruCache<H256, Bits>>,
+	jump_destinations: Mutex<MemoryLruCache<H256, CacheItem>>,
```

Comment:
Probably not in scope for this PR but I don't think the `Mutex` is necessary here.